src/Symfony/Component/FeatureFlag/Provider/InMemoryProvider.php

```diff
@@ -20,6 +20,7 @@ final class InMemoryProvider implements ProviderInterface
      */
     public function __construct(
         private readonly array $features,
+        private readonly ArgumentResolverInterface|null $argumentResolver = null,
```

Comment:
I would rather have a `ArgumentResolverAwareChecker` : allows to enable / disable this feature per project and becomes global to all providers. what about chain provider ? Shouldn't it be done recurisvely then ?

Reply:
Will an `ArgumentResolver` be used outside the `InMemoryProvider`? Argument resolving, and so TypeInfo usage, may not be necessary for some feature providers (ex: based on a database or a SaaS service).

Reply:
For some clearly it would'nt be useful. But having a way to enable it would be great eitehr through a tag (controller like) or though an interface or both.